compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -527,6 +527,12 @@ object desugar {
     // a reference to the class type bound by `cdef`, with type parameters coming from the constructor
     val classTypeRef = appliedRef(classTycon)
 
+    def applyResultTpt =
+      if isEnumCase then
+        classTypeRef
```

Comment:
Couldn't we do this unconditionally since the generated apply for a case class will infer the class type anyway?

Reply:
Yeah, that sounds good